streamdeck-ui: 2.0.6 -> 3.0.1

Majiir opened the pull request:
## Description of changes

- Added a `pythonPackages.qt6` override so that qt6 modules using are built with the correct `python` version when a Python module (e.g. `pyside6`) is built with a different Python package (e.g. `python311`).
- Added `egg_info` to `pyside6` and `shiboken6`. This is required for `streamdeck-ui` to find its `pyside6` dependency. See https://github.com/NixOS/nixpkgs/issues/167452 for background on a similar problem in `pyside2`.
- Updated upstream repository to the actively maintained fork. See: https://github.com/timothycrosley/streamdeck-ui/issues/357#issuecomment-1654606072
- Added systemd user service to the package.
- Updated version `2.0.6` -> `3.0.1`.
- Split the desktop item into one that launches normally and one with the `--no-ui` flag. The `programs.streamdeck-ui.autoLaunch` flag will launch the `--no-ui` item for consistency with current behavior. The default (visible) desktop item no longer needs `--no-ui` because the application properly handles multiple instances being launched as of version `2.0.14`.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Majiir committed:
pythonPackages.shiboken6: add egg_info

Majiir committed:
pythonPackages.pyside6: add egg_info

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/need-some-help-for-a-pr-streamdeck-ui-update/30652/4

SebTM commented:
Hey, thanks for taking care of this - I attempted it myself but wasn't nearly that successful. I tried to check it out (as it appears to be broken-build at unstable anyway atm) but I fail with the following error I can't really explain as it is in propagatedBuildInputs and I tried adding it to nativeBuildInputs as well:

```
> ERROR: Could not find a version that satisfies the requirement pillow<10.0.0,>=9.4.0 (from streamdeck-linux-gui) (from versions: none)
> ERROR: No matching distribution found for pillow<10.0.0,>=9.4.0
``` 

Maybe you can guide me some direction? :v:

Majiir commented:
`streamdeck-ui` builds fine on this branch, which is based on `master` (unstable). This PR just can't merge yet because it breaks the build for another package.

If you rebased this PR on another branch, then there's probably a dependency on `master`/unstable that you're missing.

SebTM commented on pkgs/applications/misc/streamdeck-ui/default.nix:
```diff
@@ -89,13 +78,10 @@ python3Packages.buildPythonApplication rec {
+      --ignore=tests/test_api.py
   '';
 
   meta = with lib; {
```
Just a nitpick but we should update homepage as well :v:

SebTM commented on pkgs/applications/misc/streamdeck-ui/default.nix:
```diff
@@ -3,45 +3,30 @@
 , fetchFromGitHub
 , fetchpatch
```
unused

SebTM replied:
```suggestion
```

SebTM commented on pkgs/applications/misc/streamdeck-ui/default.nix:
```diff
@@ -3,45 +3,30 @@
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     repo = pname;
```
```suggestion
    repo = "streamdeck-linux-gui";
```

Seems to has changed as well (https://github.com/streamdeck-linux-gui/streamdeck-linux-gui) wouldn't change the package-name

SebTM commented:
I figured out my issue can't explain why it works for you - I have to patch the pillow-version in pyproject.toml:

```
postPatch = ''
  substituteInPlace pyproject.toml \
    --replace 'pillow = "^9.4.0"' 'pillow = "*"'
'';
```

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/pythonpath-missing-dependencies-for-3-11-build/31620/1

Majiir committed:
pythonPackages: add qt6 override

A few qt6 modules require python. We need to override with the
appropriate python version in order to use qt6 with non-default python
packages (like python311).

Majiir committed:
streamdeck-ui: use new repo

Majiir committed:
streamdeck-ui: add systemd user service

Majiir committed:
streamdeck-ui: 2.0.6 -> 3.0.1

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/pythonpath-missing-dependencies-for-3-11-build/31620/3

Majiir committed:
streamdeck-ui: split desktop items

Majiir commented:
`nixpkgs-review` passes:

```
8 packages built:
python310Packages.pyside6 python310Packages.shiboken6 python311Packages.pyside6 python311Packages.shiboken6 smb3-foundry streamdeck-ui streamdeck-ui.dist syncplay
```

blackheaven commented:
I'm using it with `x86_64-linux`/`nixos-unstable`, no problem so far (thanks for your work)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2550

SebTM reviewed (approved):
Still running without issues :)

galexrt commented:
Can confirm as well, works fine on x86_64 nixos-unstable.

wegank commented on pkgs/development/python-modules/shiboken6/default.nix:
```diff
@@ -35,7 +34,10 @@ llvmPackages.stdenv.mkDerivation rec {
     llvmPackages.libclang
-    qt6.qtbase
+    python.pkgs.qt6.qtbase
+    python.pkgs.ninja-python
```
```suggestion
    python.pkgs.ninja
```

Majiir replied:
Fixed in 1df4a27001f8c57a639422468c70d2e7240f8349.

wegank commented on pkgs/development/python-modules/pyside6/default.nix:
```diff
@@ -34,9 +33,12 @@ stdenv.mkDerivation rec {
+  buildInputs = with python.pkgs.qt6; [
     # required
     qtbase
+    python.pkgs.ninja-python
```
```suggestion
    python.pkgs.ninja
```

Majiir replied:
Got that one too.

wegank commented:
@ofborg build python3Packages.shiboken6 python3Packages.pyside6
